### ISSUE TITLE
Failed HTTP proof after enabling HSTS

### ISSUE THREAD

**Issue by lorenzhs** [keybase/keybase-issues#662]
Hi! I recently improved my website's setup, and among others moved to a HTTPS only setup by setting the HSTS header. This means that my website is no longer reachable via http, and in fact the http version automatically redirects to https. I did a new identity proof for `web`, proving ownership of the HTTPS version. Until here, everything is fine.

Now when I want to verify my http identity proof, the keybase client receives an `HTTP 301 Moved Permanently` redirect and the proof fails:

```
$ keybase id lorenz
✔ public key fingerprint: 7871 0F91 B81A D783 A314 6481 2452 F4DF D3FC 902C
✔ "znetroll" on twitter: https://twitter.com/znetroll/status/454232484935634944
✔ "lorenzhs" on github: https://gist.github.com/10373791
✖ admin of 4z2.de via HTTP: http://4z2.de/keybase.txt (failed with code 201)
✔ admin of 4z2.de via HTTPS: https://4z2.de/keybase.txt
✔ admin of the DNS zone for 4z2.de
```

I'm not sure what the correct behaviour would be here, but this doesn't seem correct.

**Comment by Diti** [comment]
Is this issue still being looked after? I agree with @lorenzhs, there should be a way of making HSTS-enabled websites more "interesting" for Keybase. :)

**Comment by zQueal** [comment]
I think the issue here is that Keybase really doesn't care about protocol. It's just attempting to verify if you have ownership of a domain or service--it really doesn't have to care that your service __should__ only be accessible via HTTPS rather than simply HTTP. Keybase is monitoring your website via HTTP and HTTPS to see if the validation changes and technically it did. Your site is no longer reachable via HTTP.

Have you thought about doing DNS verification for HTTP/S? Either that or you can system link the two `keybase.txt` files for both HTTP and HTTPS verification.

**Comment by maxtaco** [comment]
@diti, I agree with @lorenzhs, we should indicate HSTS support.  We'll add it to the (ever-growing) list of things to do, though honestly, it's not very high priority right now.
